src/de/darmstadt/tu/crossing/CipherInputStream.cryptsl

```diff
@@ -0,0 +1,30 @@
+SPEC javax.crypto.CipherInputStream
+OBJECTS
+    byte[] b;
+    int off;
+    int len;
+    long n;
+    java.io.InputStream is;
+    javax.crypto.Cipher c;
+
+EVENTS
+    c1: CipherInputStream(is, c);
+
+    r1: read();
+    r2: read(b);
+    r3: read(b, off, len);
+    Reads := r1 | r2 | r3;
+
+    c: close();
+
+FORBIDDEN
+    CipherInputStream(is) => c1;
+
+ORDER
+    c1, Reads, c
+
+CONSTRAINTS
+    len > off;
+
+ENSURES
```

Comment:
That's not what I meant. Are you writing the rules in the CrySL editor? This should have caused parsing errors.
Here, you add 
REQUIRES
  initializedCipher[c];
In Cipher, you add under ENSURES:
  initializedCipher[this] after Inits

Same goes for the other three classes.

Reply:
Please install the CrySL editor then and let its parser check the rules before we merge the rules into the master branch. To set it up, get the [CrySL repository](https://github.com/crossingtud/cryptsl). If you need assistance, ask @AndreSonntag.

Reply:
Ok, I will do so. Thanks